### PR TITLE
T3 Audio component + analytics events

### PR DESCRIPTION
**Pull request by Raemon**
This refactors the t3 audio stuff into a separate component so that we can do audio analytics to it.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205381762406218) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/posts/PostsPage/PostsPage.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ export const getPostDescription = (post: {
   shortform: boolean;
   user: { displayName: string } | null;
 }) => {
-  if (post?.socialPreviewData?.text) {
```
Comment (Raemon) [inline packages/lesswrong/components/posts/PostsPage/PostsPage.tsx:53]: lint was yelling at me about this

File: packages/lesswrong/components/posts/PostsPage/PostsPage.tsx [hunk 2]
```diff
@@ -412,7 +412,7 @@ const PostsPage = ({post, eagerPostComments, refetch, classes}: {
     //eslint-disable-next-line react-hooks/exhaustive-deps
   }, [post._id]);
   
-  const isOldVersion = query?.revision && post.contents;
```
Comment (Raemon) [inline packages/lesswrong/components/posts/PostsPage/PostsPage.tsx:415]: lint was yelling at me about this

File: packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// TODO: Import component in components.ts
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx:1]: Delete this?

File: packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import classNames from 'classnames';
+import { useEventListener } from '../../hooks/useEventListener';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  embeddedPlayer: {
+    marginBottom: "30px"
+  },
+  hideEmbeddedPlayer: {
+    display: "none"
+  },
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx:15]: Deleted from `PostsPage`?

File: packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    </div>
+}
+
+const T3AudioPlayerComponent = registerComponent('T3AudioPlayer', T3AudioPlayer, {styles});
+
+declare global {
+  interface ComponentTypes {
+    T3AudioPlayer: typeof T3AudioPlayerComponent
+  }
+}
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx:64]: missing newline

File: packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx (hunk 4 above)
Comment (Raemon) [reply packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx:15]: These are actually still used on PostsPage (for the other audio embed)

File: packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx (hunk 5 above)
Comment (Raemon) [reply packages/lesswrong/components/posts/PostsPage/T3AudioPlayer.tsx]: fixed

**Review by b0b3rt — APPROVED** [review]
lgtm

**Pull request merged**